src/test/java/org/jenkinsci/plugins/gitclient/JGitClientTest.java

```diff
@@ -0,0 +1,162 @@
+package org.jenkinsci.plugins.gitclient;
+
+import hudson.EnvVars;
+import hudson.model.TaskListener;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitObject;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import java.util.Set;
+
+// Collides with implicit org.jenkinsci.plugins.gitclient.Git.
+//import org.eclipse.jgit.api.Git;
+import static org.eclipse.jgit.lib.Constants.HEAD;
+import org.eclipse.jgit.internal.storage.file.FileRepository;
+import org.eclipse.jgit.internal.storage.file.GC;
+import org.eclipse.jgit.lib.ObjectId;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import org.jvnet.hudson.test.Issue;
+
+/**
+ * JGit Client-specific tests. In their own Test file to ward off bloat in 
+ * GitClient and minimized parameterized and other test logic conditional 
+ * on Git implementation.
+ *
+ * @author Brian Ray
+ */
+public class JGitClientTest {
```

Comment:
Test class name is a little artificial. There's no underlying {{JGitClient}} class. But the choice of name is for symmetry with {{GitClientTest}} and for reasons of refactoring away from old JUnit 3 test classes as [discussed here](https://issues.jenkins-ci.org/browse/JENKINS-57205?focusedCommentId=385290&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-385290).

All that said I can rename it to anything that makes more sense.

Reply:
I think that `JGitClientTest` is a little too general for the specific case this is testing.  Would you consider renaming it to something like `JGitLightweightTagTest`?  We may even want to include a lightweight tag test in the CLI git code, and could reuse your JGit implementation that creates lightweight tags so that we don't need to extend the git client API to create lightweight tags.

Reply:
Ah yes you did mention over in the JIRA factoring this down to a more specific test class. I'm happy to make this and all suggested changes. Should be able to turn them around on Monday.